plugins/ocp/ocp-nvme.c

```diff
@@ -0,0 +1,672 @@
+            (uint32_t)le32_to_cpu(*(uint32_t *)&log_data[SCAO_EEDC]));
+    printf("  System data percent used			%d\n",
+            (__u8)log_data[SCAO_SDPU]);
+    printf("  Refresh counts				%"PRIu64"\n",
+            (uint64_t)(le64_to_cpu(*(uint64_t *)&log_data[SCAO_RFSC])& 0x00FFFFFFFFFFFFFF));
+    printf("  Max User data erase counts			%"PRIu32"\n",
+            (uint32_t)le32_to_cpu(*(uint32_t *)&log_data[SCAO_MXUDEC]));
+    printf("  Min User data erase counts			%"PRIu32"\n",
+            (uint32_t)le32_to_cpu(*(uint32_t *)&log_data[SCAO_MNUDEC]));
+    printf("  Number of Thermal throttling events		%d\n",
+            (__u8)log_data[SCAO_NTTE]);
+    printf("  Current throttling status		  	0x%x\n",
+            (__u8)log_data[SCAO_CTS]);
+    printf("  PCIe correctable error count			%"PRIu64"\n",
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_PCEC]));
+    printf("  Incomplete shutdowns				%"PRIu32"\n",
+            (uint32_t)le32_to_cpu(*(uint32_t *)&log_data[SCAO_ICS]));
+    printf("  Percent free blocks				%d\n",
+            (__u8)log_data[SCAO_PFB]);
+    printf("  Capacitor health				%"PRIu16"\n",
+            (uint16_t)le16_to_cpu(*(uint16_t *)&log_data[SCAO_CPH]));
+    printf("  Unaligned I/O					%"PRIu64"\n",
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_UIO]));
+    printf("  Security Version Number			%"PRIu64"\n",
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_SVN]));
+    printf("  NUSE - Namespace utilization			%"PRIu64"\n",
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_NUSE]));
+    printf("  PLP start count				%.0Lf\n",
+            int128_to_double(&log_data[SCAO_PSC]));
+    printf("  Endurance estimate				%.0Lf\n",
+            int128_to_double(&log_data[SCAO_EEST]));
+    smart_log_ver = (uint16_t)le16_to_cpu(*(uint16_t *)&log_data[SCAO_LPV]);
+    printf("  Log page version				%"PRIu16"\n",smart_log_ver);
+    printf("  Log page GUID					0x");
+    printf("%"PRIx64"%"PRIx64"\n",(uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_LPG + 8]),
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_LPG]));
+    if(smart_log_ver > 2) {
+        printf("  Errata Version Field                          %d\n",
+                (__u8)log_data[SCAO_EVF]);
+        printf("  Point Version Field                           %"PRIu16"\n",
+                (uint16_t)log_data[SCAO_PVF]);
+        printf("  Minor Version Field                           %"PRIu16"\n",
+                (uint16_t)log_data[SCAO_MIVF]);
+        printf("  Major Version Field                           %d\n",
+                (__u8)log_data[SCAO_MAVF]);
+        printf("  NVMe Errata Version				%d\n",
+                (__u8)log_data[SCAO_NEV]);
+        printf("  PCIe Link Retraining Count			%"PRIu64"\n",
+            (uint64_t)le64_to_cpu(*(uint64_t *)&log_data[SCAO_PLRC]));
+    }
+    printf("\n");
+}
+
+static void ocp_print_C0_log_json(void *data)
```

Comment:
Seems that this can be combined with the ocp_print_C0_log_normal() since almost same code as duplicated.

Reply:
Could we maybe go the other way and nest the groups in json such as -

Reply:
I see. Just considered both output functions needed to maintain as same in future.